### PR TITLE
Take EmailAddress validation out of EmailAddress

### PR DESCRIPTION
**Pull request by leogrim**
@2is10 @eishay 
Dropping the assumption that email addresses present in the database are all correct. Not the case today (recent RichSocialConnection exceptions) and possibly later on if we ever change the regex. 

It's a bit riskier but I hope that validating incoming JSON / forms / query strings parameters will cover most scenarios. 
- If our main email table is in a remote service (EmailAccount in ABook, hopefully soon), emails will always go through serialization before being persisted anyway.

### CODE REVIEW

**Comment by 2is10** [comment]
ok with me

**Comment by 2is10** [comment]
Is it possible to write an `.apply(addr: String)` method on the `EmailAddress` companion object that does validation? Then people would have to use `new EmailAddress(addr)` or a different helper method on the `EmailAddress` companion object to bypass validation. Not sure if this is a good idea... just brainstorming.

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/AuthCommander.scala [hunk 1]
```diff
@@ -52,7 +52,7 @@ case class SocialFinalizeInfo(
 
 object SocialFinalizeInfo {
   implicit val format = (
-      (__ \ 'email).format[String] and
+      (__ \ 'email).format[EmailAddress] and
```
Comment (eishay) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/AuthCommander.scala]: :star2:

**Comment by eishay** [comment]
I don't see why not keeping extra validation, it does not hurt imo. We have it in the `ExternalId` and is saved us few times.

**Comment by leogrim** [comment]
@eishay That was my initial intention as well, the problem is that it prevents us from getting any malformed data out of the database (to fix it or inactivate it).

**Comment by leogrim** [comment]
@2is10 I have thought of that, I like the case class behavior though (allowing to treat it as a String in practice, with equality and so on). If we made it normal class, would you use case insensitive equality by default?

**Comment by 2is10** [comment]
I was hoping my suggestion might work with `EmailAddress` still being a case class. If not, don’t worry about it.

**Comment by 2is10** [comment]
Looks like it wouldn’t work.

```
scala> case class EA(addr: String); object EA { def apply(addr: String) = new EA(addr + addr) }
<console>:9: error: method apply is defined twice
  conflicting symbols both originated in file '<console>'
       case class EA(addr: String); object EA { def apply(addr: String) = new EA(addr + addr) }
                  ^
```

**Comment by leogrim** [comment]
EmailAddress may be worth investing in a custom class to get both the factory method that we want and the behavior of a case class. We mostly have to override equals / hash and write an unapply method. I'm going to merge this for now but I'll think about it.

**Comment by 2is10** [comment]
Sounds good. Yes, with a custom equals, I’d compare the domains case-insensitively by default.

**Pull request merged**